composer/trainer/trainer.py

```diff
@@ -246,9 +263,13 @@ def __init__(
             precision=precision,
             precision_context=precision_context,
             train_dataloader=train_dataloader.dataloader,
-            eval_dataloader=eval_dataloader.dataloader,
+            evaluators=self.evaluators,
         )
 
+        if self.evaluators:
```

Comment:
I don't think this should need the if guard? State.evaluators is always a (potentially empty) list?